### PR TITLE
feat(jsii): support aliasing type unions

### PR DESCRIPTION
**Pull request by RomainMuller**
RFC: aws/aws-cdk-rfcs#193, aws/aws-cdk-rfcs#194

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: packages/@jsii/spec/lib/assembly.ts [hunk 1]
```diff
@@ -795,6 +797,7 @@ export enum TypeKind {
   Class = 'class',
   Enum = 'enum',
   Interface = 'interface',
+  NamedUnion = 'union',
```
Comment (rix0rrr) [inline packages/@jsii/spec/lib/assembly.ts:800]: Does this not need an assembly version bump? There will now be assemblies that old version of jsii have no idea what to do with.

File: packages/jsii-reflect/lib/assembly.ts [hunk 2]
```diff
@@ -221,7 +221,7 @@ export class Assembly extends ModuleLike {
             break;
 
           default:
-            throw new Error('Unknown type kind');
+            throw new Error(`Unsupported type kind: ${typeSpec.kind}`);
```
Comment (rix0rrr) [inline packages/jsii-reflect/lib/assembly.ts:224]: Ugh

**Comment by jsteinich** [comment]
@RomainMuller Is this something the core team is likely to pick back up anytime soon? Or could this be picked up by an external contributor (possibly myself)?
I see this as being quite important for a potential change in cdktf.

**Comment by RomainMuller** [comment]
@jsteinich as I said on Slack - I don't think we're going to be getting to this anytime soon, but I plan to get to it in 2021/2022... Community contributions are always welcome, and I'll be happy to provide pointers & guidances as is necessary if someone wants to take on the challenge.

**Comment by RomainMuller** [comment]
Discarding as stale.

**Pull request closed without merging**